Upgrade WASDK from 1.6 to 1.8

0x5bfa opened the pull request:
Closes https://github.com/CommunityToolkit/Tooling-Windows-Submodule/issues/292

0x5bfa committed:
Bump bump bump

0x5bfa committed:
Update

0x5bfa commented:
> C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(186,5): error : There is not enough space on the disk. [D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\tooling\ProjectTemplate\ProjectTemplate.sln]

The error doesn't make any sense to me.

Arlodotexe committed:
Merge branch 'main' into wasdk1.8

Arlodotexe commented:
> > C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(186,5): error : There is not enough space on the disk. [D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\tooling\ProjectTemplate\ProjectTemplate.sln]
> 
> The error doesn't make any sense to me.

The "Out of disk space" errors were caused by GitHub changing Actions from under us, this was patched in https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/296

Arlodotexe committed:
Merge branch 'main' into wasdk1.8

Arlodotexe commented:
Hmm, that's new:
```
"D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\components\CiTestExp\CiTestExp.sln" (default target) (1:2) ->
"D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\components\CiTestExp\heads\Tests.Wasdk\CiTestExp.Tests.Wasdk.csproj" (default target) (2:6) ->
(MsixCheckForInvalidCustomBeforeMicrosoftCommonTargets target) -> 
  C:\Users\runneradmin\.nuget\packages\microsoft.windows.sdk.buildtools.msix\1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.targets(33,5): error : The MSIX build tools use the 'CustomBeforeMicrosoftCommonTargets' MSBuild property to wire up a custom .props file that sets some necessary properties. This .props file does not appear to have been imported correctly. This likely means that someone has also overridden 'CustomBeforeMicrosoftCommonTargets', without ensuring to also chain the import of the previously assigned .props and .targets set to that property. You can use a binlog to learn more about where that assignment was done (set the verbosity to diagnostics, and search for 'reassignment: $(CustomBeforeMicrosoftCommonTargets'). [D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\components\CiTestExp\heads\Tests.Wasdk\CiTestExp.Tests.Wasdk.csproj]


"D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\components\CiTestExp\CiTestExp.sln" (default target) (1:2) ->
"D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\components\CiTestExp\heads\Wasdk\CiTestExp.Wasdk.csproj" (default target) (3:6) ->
(_ValidatePresenceOfAppxManifestItems target) -> 
  C:\Users\runneradmin\.nuget\packages\microsoft.windows.sdk.buildtools.msix\1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.Packaging.targets(3024,5): error MSB4062: The "Microsoft.Build.Msix.AppxManifest.WinAppSdkValidateAppxManifestItems" task could not be loaded from the assembly C:\Users\runneradmin\.nuget\packages\microsoft.windowsappsdk\1.6.250108002\buildTransitive\..\tools\net472\Microsoft.Windows.SDK.BuildTools.MSIX.dll. Could not load file or assembly 'file:///C:\Users\runneradmin\.nuget\packages\microsoft.windowsappsdk\1.6.250108002\tools\net472\Microsoft.Windows.SDK.BuildTools.MSIX.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\Tooling-Windows-Submodule\Tooling-Windows-Submodule\components\CiTestExp\heads\Wasdk\CiTestExp.Wasdk.csproj]
```

Errors from the latest CI run.

I notice it's coming from `C:\Users\runneradmin\.nuget\packages\microsoft.windowsappsdk\1.6.250108002\buildTransitive\..\tools\net472\Microsoft.Windows.SDK.BuildTools.MSIX.dll.`, so there might still be a version number to update somewhere.

We don't depend on Win2D here, but it might be coming from Behaviors.

Arlodotexe commented:
Behaviors was updated with Wasdk 1.8 last month and the update is available [on NuGet](https://www.nuget.org/packages/Microsoft.Xaml.Behaviors.WinUI.Managed/3.0.1#dependencies-body-tab), so we'll update that and see if the error goes away.

Arlodotexe committed:
Bump Microsoft.Xaml.Behaviors.WinUI.Managed to 3.0.1

michael-hawker commented:
@Arlodotexe we should be able to move this along now that Win2D has updated. Can we get test PRs open on the main and labs repo for this too?

Arlodotexe committed:
Update WASDK version to 1.8.260204000 to match Win2d dependency and fix build errors

Arlodotexe committed:
Remove duplicate Microsoft.Web.WebView2 package reference, now referenced transitively from Microsoft.WindowsAppSDK.WinUI

Arlodotexe committed:
Enable diagnostics in CI for binlog output to investigate MSIX build tools issue

Arlodotexe committed:
Fix quoting issue for component path preventing -UseDiagnostics flag from working

Arlodotexe committed:
Use quotes around componentPath in GenerateSingleSampleHeads.ps1 to work around slngen issues

Arlodotexe committed:
Name artifacts uniquely based on job matrix

Arlodotexe committed:
Disable consolelogger due to CI issues

Full error text:
```
Microsoft.Build.Exceptions.InternalLoggerException: The build stopped unexpectedly because the "ReusableLogger" logger failed unexpectedly during shutdown. ---> System.ObjectDisposedException: Cannot write to a closed TextWriter.
   at System.IO.__Error.WriterClosed()
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at Microsoft.Build.Framework.BuildFinishedEventHandler.Invoke(Object sender, BuildFinishedEventArgs e)
   at Microsoft.Build.Logging.EventArgsDispatcher.Dispatch(BuildEventArgs buildEvent)
   at Microsoft.VisualStudio.SlnGen.ForwardingLogger.Shutdown()
   at Microsoft.Build.BackEnd.Logging.LoggingService.ShutdownLogger(ILogger logger)
```

Arlodotexe committed:
Align slngen command formatting with GenerateAllSolution.ps1

Arlodotexe committed:
Add PSNativeCommandArgumentPassing, further align with GenerateAllSolution.ps1

Arlodotexe committed:
Align $sdkoptions to GenerateAllSolution.ps1

Arlodotexe committed:
Add matrix multitarget to artifact name to fix conflicts

Arlodotexe committed:
Add msbuild binlog and diagnostic params, realign artifact name with mainline repo

Arlodotexe committed:
Set MSBUILD_VERBOSITY to diagnostic

Arlodotexe commented:
I managed to produce the binlog for this error we were getting:
```
The MSIX build tools use the 'CustomBeforeMicrosoftCommonTargets' MSBuild property to wire up a custom .props file that sets some necessary properties. This .props file does not appear to have been imported correctly. This likely means that someone has also overridden 'CustomBeforeMicrosoftCommonTargets', without ensuring to also chain the import of the previously assigned .props and .targets set to that property. You can use a binlog to learn more about where that assignment was done (set the verbosity to diagnostics, and search for 'reassignment: $(CustomBeforeMicrosoftCommonTargets').
```

Verbosity set to diagnostic (as suggested by the error), we're not seeing anything for this reassignment in the binlog:

<img width="656" height="386" alt="image" src="https://github.com/user-attachments/assets/fb4cc3c6-e1a0-4679-99a3-efe7b9d13aaf" />

<img width="756" height="211" alt="image" src="https://github.com/user-attachments/assets/d7b5636b-648f-4359-9915-325decc695da" />

It's unclear why this error is surfacing with no clear path for investigation.

Suggestions welcome.

michael-hawker commented:
~~@Arlodotexe where's the update to the Win2D version? I don't see that in the changes?~~

NVM, that's the other repos only I guess, so something else here.

Arlodotexe committed:
Bumped Microsoft.Xaml.Behaviors.Uwp.Managed to latest 3.0.1

michael-hawker reviewed (approved):
Approving as we know this works E2E with the mainline PR now passing, we just have the circular dependency problem to align on how to coordinate.

@Arlodotexe I assume we haven't hit this quite the same way before as this is a larger breaking change with the WASDK 1.8 update than we've had in the past. We probably had the same issues, but it just didn't cause an issue with the mismatch while we were in transition.

Arlodotexe committed:
Updated WCT Extensions reference to 8.3.260331-pull-837.1767

Arlodotexe committed:
Update gallery app package references to 8.3.260331-pull-837.1767

Arlodotexe committed:
Temp add MainLatest and PullRequest feeds, prerelease toolkit 8.3 packages with Microsoft.WindowsAppSdk 1.8 upgrade

michael-hawker commented on .github/workflows/build.yml:
```diff
@@ -16,7 +16,8 @@ on:
 
 env:
-  ENABLE_DIAGNOSTICS: false
+  ENABLE_DIAGNOSTICS: true
+  MSBUILD_VERBOSITY: diagnostic
```
@Arlodotexe we probably don't want these on by default, we should update to use the flag system to be able to enable these in manual runs in the UI like I did for XAML Studio.
